lib/gitlab_api/base.rb

```diff
@@ -0,0 +1,46 @@
+module GitlabApi
+  class Base
+    attr_accessor :response
+
+    def self.perform(*attrs)
+      new(*attrs).perform
+    end
+
+    def connection
+      @connection ||= Faraday.new(url: gitlab_api_url) do |faraday|
+        faraday.headers['Content-Type'] = 'application/json'
+        faraday.headers['Accept'] = 'application/json'
+        faraday.headers['Private-Token'] = gitlab_private_token
+        faraday.adapter Faraday.default_adapter
+      end
+    end
+
+    def get(path)
+      self.response = connection.get(path)
+      self
+    end
+
+    def post(&block)
+      self.response = connection.post(&block)
+      self
+    end
+
+    def data
+      @data ||= JSON.parse(response&.body)
+    end
+
+    def success?
+      response.status == self.class::SUCCESS_RESPONSE_CODE
```

Comment:
I've never used Faraday before, but can't you use this instead? https://www.rubydoc.info/gems/faraday/Faraday%2FResponse:success%3F

Reply:
Since gitlab's api is somewhat inconsistent within itself about the response status, it seemed to be better to do it this way in order to prevent an false positive. The constant is what we expect from the API, anything else should be considered an failure, the faraday method considers any "success" response code as a success, which might not be the case here.